### PR TITLE
Add pre-compiled headers to one of the CI runners

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #82820
* __->__ #77351

Pre-compiled headers currently aren't tested in CI.

### CODE REVIEW

**Review by janeyx99 — COMMENTED** [review]
This looks fine on the technical side. 

Question: How did you choose the xenial, python 3.7, gcc7 build?

**Comment by peterbell10** [comment]
I picked that particular job mainly because it's a plain CPU job and also the `gcc5.4` and `clang` builds had larger test matrices, so when the PCH build gets broken they would lose more CI signal if I changed them.

**Review by janeyx99 — APPROVED** [review]

File: .github/merge_rules.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
          "linux-xenial-py3.7-gcc7-no-ops / build",
          "linux-bionic-py3.7-clang9 / build",
          "linux-xenial-py3.7-clang7-onnx / build",
-         "linux-xenial-py3.7-gcc7 / build",
+         "linux-xenial-py3.7-gcc7-pch / build",
```
Comment (peterbell10) [inline .github/merge_rules.json:31]: @janeyx99 is this right?

**Pull request closed without merging**